packages/libs/error-stack/examples/demo.rs

```diff
@@ -16,14 +16,33 @@ impl fmt::Display for ParseExperimentError {
 
 impl Context for ParseExperimentError {}
 
-fn parse_experiment(description: &str) -> Result<(u64, u64), ParseExperimentError> {
-    let value = description
-        .parse()
-        .into_report()
-        .attach_printable_lazy(|| format!("{description:?} could not be parsed as experiment"))
+fn parse_experiment(description: &str) -> Result<Vec<(u64, u64)>, ParseExperimentError> {
+    let values = description
+        .split(' ')
+        .map(|value| {
+            value
+                .parse::<u64>()
+                .into_report()
+                .attach_printable_lazy(|| format!("{value:?} could not be parsed as experiment"))
+        })
+        .map(|value| value.map(|ok| (ok, 2 * ok)))
+        .fold(Ok(vec![]), |acc, value| match (acc, value) {
+            (Ok(mut acc), Ok(value)) => Ok({
+                acc.push(value);
+
+                acc
+            }),
+            (Ok(_), Err(err)) => Err(err),
+            (Err(acc), Ok(_)) => Err(acc),
+            (Err(mut acc), Err(err)) => Err({
+                acc.extend_one(err);
+
+                acc
+            }),
```

Comment:
I think this is more readable
```suggestion
            (Ok(mut acc), Ok(value)) => {
                acc.push(value);

                Ok(acc)
            },
            (Ok(_), Err(err)) => Err(err),
            (Err(acc), Ok(_)) => Err(acc),
            (Err(mut acc), Err(err)) => {
                acc.extend_one(err);

                Err(acc)
            },
```

Reply:
fixed in [d1cf758](https://github.com/hashintel/hash/pull/1101/commits/d1cf758932175d95247ed717ca86bc19d0f628ac)